src/python/fuzzing/coverage_uploader.py

```diff
@@ -86,9 +83,24 @@ def upload_testcases_if_needed(fuzzer_name, testcase_list, testcase_directory):
   gcs_base_url += utils.string_hash(identifier)
 
   list_gcs_url = gcs_base_url + '/' + LIST_FILE_BASENAME
-  if not storage.write_data('\n'.join(testcase_list), list_gcs_url):
+  if not storage.write_data('\n'.join(files_list), list_gcs_url):
     return
 
-  runner.rsync(testcase_directory, gcs_base_url)
-  logs.log('Synced {count} test cases to {gcs_url}'.format(
-      count=len(testcase_list), gcs_url=gcs_base_url))
+  if has_testcases_in_testcase_directory:
+    # Sync everything in |testcase_directory| since it is fuzzer-generated.
+    runner.rsync(testcase_directory, gcs_base_url)
+
+  if has_testcases_in_data_directory:
+    # Sync all fuzzer generated testcase in data bundle directory.
+    runner.rsync(
+        data_directory,
+        gcs_base_url,
+        exclusion_pattern=('"(^|.*/)(?!{fuzz_prefix})[^/]+$"'.format(
```

Comment:
Would this work on Windows bots (if we ever upload anything from them)?

Reply:
Nope, thanks for bringing this up. Simplifying it to work on both.